File: 100-Days-Series/Day 75/RemoveDuplicatesfromSortedList.java
class LinkedList{
    int data;
    LinkedList next;
    public LinkedList(int data) {
        this.data=data;
        this.next=null;
    }
}
public class RemoveDuplicatesfromSortedList {
    private static LinkedList head=null;// Head of the Linked List
    public static java.util.Scanner sc=new java.util.Scanner(System.in);
    // Method to add New Node 
    public static void addNode(){
        System.out.print("Enter Data : ");
        int data=sc.nextInt();
        LinkedList newNode=new LinkedList(data);
        if(head==null){
            head=newNode;
        }else{
            LinkedList temp,last=head;
            for(temp=head;temp!=null;temp=temp.next){
                last=temp;
            }
            last.next=newNode;
        }
    }
    // Method for traverse
    public static void traverse(){
        if(head==null){
            System.out.println("Linked List is Empty");
        }else{
            LinkedList temp;
            for(temp=head;temp!=null;temp=temp.next){
                System.out.println("Data is "+temp.data);
                System.out.println("------------------------------");
            }
        }
    }
    // Method to remove Duplicates
    public static void getRemoveDuplicates(){
        if(head==null){
            System.out.println("Linked List is Empty");
        }else{
            LinkedList current=head;
            while(current.next!=null){
                if(current.data==current.next.data){
                    current.next=current.next.next;
                }else{
                    current=current.next;
                }
            }
        }
    }
    // Main Method 
    public static void main(String [] args){
        // Let's Start Here
        int ch;
        do { 
            System.out.println("1. Add New Node");
            System.out.println("2. Traverse");
            System.out.println("3. Remove Duplicates");
            System.out.println("4. Exit");
            System.out.print("Select a choice : ");
            ch=sc.nextInt();
            if(ch==1){
                addNode();
            }else if(ch==2){
                traverse();
            }else if(ch==3){
                getRemoveDuplicates();
            }else if(ch==4){
                System.out.println("Exiting...");
            }else{
                System.out.println("Invalid choice");
            }
        } while (ch!=4);
    }
}

File: 100-Days-Series/Day 75/RemoveDuplicatesfromSortedList.py
class LinkedList:
    def __init__(self,data):
        self.data=data
        self.next=None

class RemoveDuplicatesfromSortedList:
    def __init__(self):
        self.head=None

    # Method for add new Node
    def add_node(self):
        data=int(input('Enter data '))
        newNode=LinkedList(data)
        if self.head==None:
            self.head=newNode
        else:
            current,last = self.head,self.head
            while current!=None:
                last=current
                current=current.next
            last.next=newNode
    # Method for traverse
    def traverse(self):
        if self.head==None:
            print('Linked List is empty')
        else:
            current=self.head
            while current!=None:
                print('Data is ',current.data)
                print('-----------------------')
                current=current.next
    # Method to remove Duplicates Node
    def get_remove_duplicates(self):
        if self.head==None:
            print('Linked list is empty')
        else:
            current=self.head
            while current.next!=None:
                if current.data==current.next.data:
                    current.next=current.next.next
                else:
                    current=current.next

# Let's Start Here
list=RemoveDuplicatesfromSortedList()
while(True):
    print('1. Add New Node')
    print('2. Traverse')
    print('3. Remove Duplicate')
    print('4. Exit')
    ch=int(input('Select a choice : '))
    if ch==1:
        list.add_node()
    elif ch==2:
        list.traverse()
    elif ch==3:
        list.get_remove_duplicates()
    elif ch==4:
        print('Exiting...')
        break
    else:
        print('Invalid choice')